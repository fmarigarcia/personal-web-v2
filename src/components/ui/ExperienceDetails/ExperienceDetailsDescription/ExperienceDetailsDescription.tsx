import React from 'react';
import { useTranslation } from 'react-i18next';

interface ExperienceDetailsDescriptionProps {
    experienceId: string;
    descriptionKeys: string[];
}

export const ExperienceDetailsDescription: React.FC<
    ExperienceDetailsDescriptionProps
> = ({ experienceId, descriptionKeys }) => {
    const { t } = useTranslation();

    return (
        <ul className="space-y-4 mb-8">
            {descriptionKeys.map((descKey, i) => (
                <li key={`${experienceId}-${i}`} className="flex items-start">
                    <div className="w-2 h-2 bg-stone-950 dark:bg-green-400 rounded-full mt-2 mr-4 flex-shrink-0" />
                    <span className="text-stone-700 dark:text-zinc-300 leading-relaxed">
                        {t(descKey)}
                    </span>
                </li>
            ))}
        </ul>
    );
};
