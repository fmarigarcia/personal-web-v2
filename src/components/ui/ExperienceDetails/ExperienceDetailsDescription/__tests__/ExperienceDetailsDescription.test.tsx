import { render, screen } from '@testing-library/react';
import { ExperienceDetailsDescription } from '../ExperienceDetailsDescription';

// Mock react-i18next
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: jest.fn((key: string) => key),
    }),
}));

describe('ExperienceDetailsDescription', () => {
    const defaultProps = {
        experienceId: 'exp-1',
        descriptionKeys: [
            'experience.exp1.desc1',
            'experience.exp1.desc2',
            'experience.exp1.desc3',
        ],
    };

    it('should render all description items', () => {
        render(<ExperienceDetailsDescription {...defaultProps} />);

        expect(screen.getByText('experience.exp1.desc1')).toBeInTheDocument();
        expect(screen.getByText('experience.exp1.desc2')).toBeInTheDocument();
        expect(screen.getByText('experience.exp1.desc3')).toBeInTheDocument();
    });

    it('should render as an unordered list', () => {
        render(<ExperienceDetailsDescription {...defaultProps} />);

        const list = screen.getByRole('list');
        expect(list.tagName).toBe('UL');
        expect(list).toHaveClass('space-y-4', 'mb-8');
    });

    it('should render correct number of list items', () => {
        render(<ExperienceDetailsDescription {...defaultProps} />);

        const listItems = screen.getAllByRole('listitem');
        expect(listItems).toHaveLength(3);
    });

    it('should have correct list item structure', () => {
        render(<ExperienceDetailsDescription {...defaultProps} />);

        const listItems = screen.getAllByRole('listitem');
        listItems.forEach(item => {
            expect(item).toHaveClass('flex', 'items-start');
        });
    });

    it('should render bullet points for each item', () => {
        const { container } = render(<ExperienceDetailsDescription {...defaultProps} />);

        const bullets = container.querySelectorAll('.w-2.h-2.bg-blue-600.rounded-full');
        expect(bullets).toHaveLength(3);
    });

    it('should style description text correctly', () => {
        const { container } = render(<ExperienceDetailsDescription {...defaultProps} />);

        const textSpans = container.querySelectorAll('span.text-gray-700.leading-relaxed');
        expect(textSpans).toHaveLength(3);
    });

    it('should render correct number of list items', () => {
        const { container } = render(<ExperienceDetailsDescription {...defaultProps} />);

        const listItems = container.querySelectorAll('li');
        expect(listItems).toHaveLength(3);
        expect(listItems[0]).toHaveTextContent('experience.exp1.desc1');
        expect(listItems[1]).toHaveTextContent('experience.exp1.desc2');
        expect(listItems[2]).toHaveTextContent('experience.exp1.desc3');
    });

    it('should handle empty description keys', () => {
        render(<ExperienceDetailsDescription {...defaultProps} descriptionKeys={[]} />);

        const list = screen.getByRole('list');
        expect(list).toBeInTheDocument();
        
        const listItems = screen.queryAllByRole('listitem');
        expect(listItems).toHaveLength(0);
    });

    it('should handle single description key', () => {
        render(
            <ExperienceDetailsDescription
                {...defaultProps}
                descriptionKeys={['single.description']}
            />
        );

        expect(screen.getByText('single.description')).toBeInTheDocument();
        
        const listItems = screen.getAllByRole('listitem');
        expect(listItems).toHaveLength(1);
    });

    it('should translate all description keys', () => {
        render(<ExperienceDetailsDescription {...defaultProps} />);

        // Verifies that t() is called for each description key
        expect(screen.getByText('experience.exp1.desc1')).toBeInTheDocument();
        expect(screen.getByText('experience.exp1.desc2')).toBeInTheDocument();
        expect(screen.getByText('experience.exp1.desc3')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<ExperienceDetailsDescription {...defaultProps} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with empty descriptions', () => {
        const { container } = render(
            <ExperienceDetailsDescription
                {...defaultProps}
                descriptionKeys={[]}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
