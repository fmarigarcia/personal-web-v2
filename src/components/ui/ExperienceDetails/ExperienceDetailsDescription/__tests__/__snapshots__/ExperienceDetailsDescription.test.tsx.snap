// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceDetailsDescription should match snapshot 1`] = `
<ul
  class="space-y-4 mb-8"
>
  <li
    class="flex items-start"
  >
    <div
      class="w-2 h-2 bg-stone-950 dark:bg-green-400 rounded-full mt-2 mr-4 flex-shrink-0"
    />
    <span
      class="text-stone-700 dark:text-zinc-300 leading-relaxed"
    >
      experience.exp1.desc1
    </span>
  </li>
  <li
    class="flex items-start"
  >
    <div
      class="w-2 h-2 bg-stone-950 dark:bg-green-400 rounded-full mt-2 mr-4 flex-shrink-0"
    />
    <span
      class="text-stone-700 dark:text-zinc-300 leading-relaxed"
    >
      experience.exp1.desc2
    </span>
  </li>
  <li
    class="flex items-start"
  >
    <div
      class="w-2 h-2 bg-stone-950 dark:bg-green-400 rounded-full mt-2 mr-4 flex-shrink-0"
    />
    <span
      class="text-stone-700 dark:text-zinc-300 leading-relaxed"
    >
      experience.exp1.desc3
    </span>
  </li>
</ul>
`;

exports[`ExperienceDetailsDescription should match snapshot with empty descriptions 1`] = `
<ul
  class="space-y-4 mb-8"
/>
`;
